### ISSUE TITLE
update document

### ISSUE THREAD

**Issue by serkandemirel0420** [blevesearch/bleve#1427]
is it possible to update indexed document, also realised when deleted a document the size of the index does not reduce(it just adds new flag* to it).

**Comment by mschoch** [comment]
> is it possible to update indexed document

Yes, whenever you call `Index()` either directly on the index, or a `Batch`, the contents replace whatever was indexed with that identifier before.

> also realised when deleted a document the size of the index does not reduce(it just adds new flag* to it)

This is not really true, but the details vary from by index implementation.  If you are using upside_down (the default, but not recommended), then document data is deleted, but it is true that some information in the term dictionary remains forever.  If you are using scorch, then document data is deleted eventually, but since all documents are stored in immutable segment files, it cannot be reclaimed until that segment is rewritten during a merge operation.  How aggressively we merge based on deleted documents is somewhat tune-able.

If you can describe more about what you're trying to do we can try to offer more specific suggestions.

**Comment by serkandemirel0420** [comment]
thank you but i would not know where to start and what to ask.

this is one of the best library i have seen in a long time but unfortunately i can not say the same thing about the documentation, sample apps and even the amount of the comment in the repository. I have tried to use it, come up to a point but i am moving very slow and also as i move forward i'am having more questions about what have i done. 

please consider to focus on documentation and sample apps.

**Comment by serkandemirel0420** [comment]
I am trying to index json files using upside_down. Files are changing constantly and the index size keeps growing. even when i delete all the documents from the index, the file size of the index remains the same. And this is a huge problem in my case. That is basically it.

**Comment by mschoch** [comment]
Is there a reason you're using upsidedown?  We no longer recommend it, it is the default only for backwards compatibility reasons, and we plan to make scorch the default in the 2.0 release.

There are issues with reclaiming space over time with BoltDB, the default K/V store for upsidedown.  I think some users had success accessing the underlying database and calling Compact on it, but honestly it's been a long time and I'm not sure.

Deleting everything and reclaiming all space isn't a use case we've focused on, so it's possible none of the options will work well for you.  But I would suggest at least trying scorch, as we are likely to offer better support.
